quintilesims/layer0
# [WIP] 274: Environment Scaler

**What does this pull request do?**
- Scaler no longer stops counting a provider as a provider if the agent is disconnected.
- Scaler selects the most efficient method of resource consumer allocation by comparing sort-by-memory and sort-by-CPU.
- Scaler aborts scaling actions if they would increase/decrease the number of instances above/below `max_count`/`min_count`.


**How should this be tested?**
Do some scaling operations (you'll want to use the dev CLI in order to have access to the `environment set-scale` command) and confirm that the API handles them properly (debug statements may be monitored by running the API locally with the `-debug` flag). _Be especially on the lookout for any mismatch or weirdness stemming from the fact that the scaler no longer looks at pending ECS tasks when gathering resource consumers._


**Checklist**


Part of work to address #274; also closes #421
